KAN-234-엘라스틱서치-패키지-분리-리팩토링

llRosell opened the pull request:
## #️⃣ CHEERHA Board Number

<!--- ex) #이슈번호, #이슈번호, 또는 release version -->
https://spring-3-jo.atlassian.net/browse/KAN-234

## 📝 요약

<!--- 변경 사항 및 관련 이슈에 대해 간단하게 작성해주세요. 어떻게보다 무엇을 왜 수정했는지 설명해주세요. -->
엘라스틱서치 채용공고 서비스 리팩토링: 코드 중복 제거 및 필터링 로직 분리

calculateFrom 메서드 및 buildSearchRequest 메서드를 프라이빗 메서드로 분리하여 중복 코드 제거
필터링 로직을 JobOpeningDocumentFilter 클래스로 분리하여 가독성 및 유지보수성 개선
ReadJobOpeningElasticResponseDto에서 DTO 변환 로직을 toDto 메서드로 분리하여 처리
마감된 채용공고 보이지 않도록 필터링
Elasticsearch 관련 쿼리 실행 로직을 `ElasticsearchClientService`로 분리
마감된 채용공고 필터링을 상수 `CURRENT_DATE_TIME`을 사용하여 가독성 향상

## 🛠️ PR 유형

- [ ] 배포용 PR : dev 테스트를 완료하였나요?
- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [x] 코드 리팩토링
- [x] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## 💬 논의사항 or 질문

<!--- 리뷰어가 중점적으로 봐줬으면 좋겠는 부분이 있으면 적어주세요. -->
<!--- 논의해야할 부분이 있다면 적어주세요.-->
<!--- ex) 메서드 XXX의 이름을 더 잘 짓고 싶은데 혹시 좋은 명칭이 있을까요? -->

## ✅ PR Checklist

PR이 다음 요구 사항을 충족했는지 확인하기!

- [ ] 커밋 메시지 컨벤션 준수
- [ ] 변경 사항 테스트 여부 체크(버그 수정/기능 테스트)

## 💥 리뷰어 지목!!!!!

<!--- 리뷰어를 2명 이상 지목해주세요. -->
- [x] 채영
- [x] 지현
- [ ] 리은
- [x] 승찬
- [ ] 주양

llRosell committed:
♻️ refactor : 코드 중복 제거 및 필터링 로직 분리
calculateFrom 메서드 및 buildSearchRequest 메서드를 프라이빗 메서드로 분리하여 중복 코드 제거
필터링 로직을 JobOpeningDocumentFilter 클래스로 분리하여 가독성 및 유지보수성 개선
ReadJobOpeningElasticResponseDto에서 DTO 변환 로직을 toDto 메서드로 분리하여 처리
마감된 채용공고 보이지 않도록 필터링

llRosell committed:
♻️ refactor : 마감된 채용공고 보이지 않도록 필터링

llRosell committed:
Merge branch 'dev' of https://github.com/cheer-ha/cheer-ha into refactor/jobopening-elasticsearch

roqkfchqh commented on src/main/java/com/project/cheerha/common/exception/client/ClientErrorCode.java:
```diff
@@ -24,8 +24,7 @@ public enum ClientErrorCode {
     MAX_FAILURE_COUNT_EXCEEDED(HttpStatus.BAD_REQUEST, "이메일 토큰 인증 횟수가 초과되었습니다."),
     EMAIL_NOT_SENT_YET(HttpStatus.BAD_REQUEST, "이메일을 보낸 적이 없습니다"),
-    INVALID_PASSWORD_RESET_TOKEN(HttpStatus.BAD_REQUEST, "패스워드 변경용 토큰이 유효하지 않습니다."),
-    EXPIRED_JOB_OPENING(HttpStatus.BAD_REQUEST, "마감된 채용공고입니다.");
```
마감된 채용공고 빠지면 안 되지 않나영? 빠져도 되나요? 확인부탁드려요!

backswan0 replied:
저도 이 부분 확인 부탁드리겠습니다:)

backswan0 commented on src/main/java/com/project/cheerha/domain/elasticsearch/dto/response/ReadJobOpeningElasticResponseDto.java:
```diff
@@ -26,5 +29,35 @@ public record ReadJobOpeningElasticResponseDto(
         Integer viewCount, // 조회수
         List<String> requiredSkills // 요구되는 기술 목록
 ) {
```
List이니까 'requiredSkillList'라고 고치면 어떨까요?:)

backswan0 commented on src/main/java/com/project/cheerha/domain/elasticsearch/dto/response/ReadJobOpeningElasticResponseDto.java:
```diff
@@ -26,5 +29,35 @@ public record ReadJobOpeningElasticResponseDto(
+     * @return 변환된 `ReadJobOpeningElasticResponseDto` 객체의 리스트
+     */
+    public static List<ReadJobOpeningElasticResponseDto> toDto(List<JobOpeningDocument> jobOpeningDocuments) {
+        return jobOpeningDocuments.stream()
```
(1) List이니까 'jobOpeningDocumentList'로 고치면 어떨까요?:)
(2) 여기서 document가 정확히 무엇을 의미하나요?:) 어떤 의미로 쓰였는지 궁금합니다:)

roqkfchqh replied:
document는 엘라스틱서치에서 사용하는 관용적인 말입니당

backswan0 replied:
(2) 확인했습니다:) 감사합니다:)

backswan0 commented on src/main/java/com/project/cheerha/domain/jobopening/service/JobOpeningService.java:
```diff
@@ -122,6 +129,29 @@ public Page<ReadJobOpeningResponseDto> readTop100PopularJobOpenings(Pageable pag
+        // 마감된 채용공고를 제외하는 로직
+        ZonedDateTime now = ZonedDateTime.now();
+        List<ReadJobOpeningResponseDto> filteredDtos = dtoPage.getContent().stream()
+                .filter(dto -> dto.getHiringEndAt().isAfter(now))
```
List이니까 'filteredDtos' 대신  'readJobOpeningResponseDtoList' 또는 'filteredDtoList'로 바꾸면 어떨까요?:)

backswan0 commented on src/main/java/com/project/cheerha/domain/jobopening/service/JobOpeningService.java:
```diff
@@ -122,6 +129,29 @@ public Page<ReadJobOpeningResponseDto> readTop100PopularJobOpenings(Pageable pag
+     */
+    private Page<ReadJobOpeningResponseDto> filterExpiredJobOpenings(Page<ReadJobOpeningResponseDto> dtoPage) {
+        // 마감된 채용공고를 제외하는 로직
+        ZonedDateTime now = ZonedDateTime.now();
```
now도 괜찮은데, 이 시간이 정확히 어떤 시간인지 의미가 드러나도록 변수명을 좀 더 다듬으면 어떨까요?:)

backswan0 reviewed (commented):
고생하셨습니다:) 코멘트 확인 부탁드리겠습니다:)

roqkfchqh commented on src/main/java/com/project/cheerha/domain/jobopening/service/JobOpeningService.java:
```diff
@@ -122,6 +129,29 @@ public Page<ReadJobOpeningResponseDto> readTop100PopularJobOpenings(Pageable pag
+    /**
+     * 마감된 채용공고를 제외하는 메서드
+     */
+    private Page<ReadJobOpeningResponseDto> filterExpiredJobOpenings(Page<ReadJobOpeningResponseDto> dtoPage) {
```
이 필터도 필터클래스로 빼도 될거같습니다 !

roqkfchqh replied:
아 아니다.. 잘 모르겠네요 같은 클래스 내부의 private메서드에서 사용하는거였군요

roqkfchqh commented on src/main/java/com/project/cheerha/domain/jobopening/service/JobOpeningService.java:
```diff
@@ -14,13 +14,15 @@
+import java.util.stream.Collectors;
 
 
 @Service
```
깃허브에선 안 보이는데, ElasticSearchClient 호출부를 다른 클래스로 빼면 더 깔끔할 거 같습니다

roqkfchqh replied:
![스크린샷 2025-03-05 오전 11 45 47](https://github.com/user-attachments/assets/50524fd1-eee9-432c-ad87-87ad7a9359a7)
약간 이런 느낌으로용 (클래스명은 대강 지었씁니다)

roqkfchqh replied:
![스크린샷 2025-03-05 오전 11 46 13](https://github.com/user-attachments/assets/ef1c5c54-cbce-4959-a35e-26a849cacbef)
![스크린샷 2025-03-05 오전 11 46 22](https://github.com/user-attachments/assets/7ba92fce-12fd-4c20-a55a-f3b821d55842)
ㅇ이렇게 사용하도록!

tmdcksdl replied:
따로 어떻게 빼면 좋을지 고민하고 있었는데, 저렇게 다른 클래스로 분리하니까 코드가 더 깔끔해 보이네요!

tmdcksdl commented:
코드 확인했습니다!
지현님이 말씀해주신 변수명 변경과 
elasticsearchClient 호출을 다른 클래스에 분리하는 것도 괜찮은 것 같아요!

llRosell committed:
♻️ refactor : Elasticsearch 클라이언트 호출부 클래스 분리
- Elasticsearch 관련 쿼리 실행 로직을 `ElasticsearchClientService`로 분리
- 마감된 채용공고 필터링을 상수 `CURRENT_DATE_TIME`을 사용하여 가독성 향상

llRosell committed:
♻️ refactor : 명칭을 List로 수정

llRosell committed:
Merge branch 'dev' of https://github.com/cheer-ha/cheer-ha into refactor/jobopening-elasticsearch

llRosell commented:
> ## #️⃣ CHEERHA Board Number
> https://spring-3-jo.atlassian.net/browse/KAN-234
> 
> ## 📝 요약
> 엘라스틱서치 채용공고 서비스 리팩토링: 코드 중복 제거 및 필터링 로직 분리
> 
> calculateFrom 메서드 및 buildSearchRequest 메서드를 프라이빗 메서드로 분리하여 중복 코드 제거 필터링 로직을 JobOpeningDocumentFilter 클래스로 분리하여 가독성 및 유지보수성 개선 ReadJobOpeningElasticResponseDto에서 DTO 변환 로직을 toDto 메서드로 분리하여 처리 마감된 채용공고 보이지 않도록 필터링
> 
> ## 🛠️ PR 유형
> * [ ]  배포용 PR : dev 테스트를 완료하였나요?
> * [x]  새로운 기능 추가
> * [ ]  버그 수정
> * [ ]  코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
> * [x]  코드 리팩토링
> * [x]  주석 추가 및 수정
> * [ ]  문서 수정
> * [ ]  테스트 추가, 테스트 리팩토링
> * [ ]  빌드 부분 혹은 패키지 매니저 수정
> * [ ]  파일 혹은 폴더명 수정
> * [ ]  파일 혹은 폴더 삭제
> 
> ## 💬 논의사항 or 질문
> ## ✅ PR Checklist
> PR이 다음 요구 사항을 충족했는지 확인하기!
> 
> * [ ]  커밋 메시지 컨벤션 준수
> * [ ]  변경 사항 테스트 여부 체크(버그 수정/기능 테스트)
> 
> ## 💥 리뷰어 지목!!!!!
> * [x]  채영
> * [x]  지현
> * [ ]  리은
> * [x]  승찬
> * [ ]  주양
